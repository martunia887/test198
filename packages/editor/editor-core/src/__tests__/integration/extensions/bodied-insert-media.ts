import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  expectMatchDocument,
  fullpage,
  insertBlockMenuItem,
  insertMedia,
} from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';

BrowserTestCase(
  `bodied-insert-media.ts: Bodied Extension: Insert Media`,
  { skip: ['edge', 'ie', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: 'full-page',
      media: {
        allowMediaSingle: true,
        allowMediaGroup: true,
      },
      allowExtension: {
        allowBreakout: true,
      },
    });

    await page.waitForSelector(fullpage.placeholder);
    await page.click(fullpage.placeholder);

    await insertBlockMenuItem(page, 'Bodied macro (EH)');
    await insertMedia(page);

    await expectMatchDocument(page, testName);
  },
);
