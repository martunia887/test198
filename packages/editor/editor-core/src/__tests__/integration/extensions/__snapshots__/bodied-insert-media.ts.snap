// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bodied-insert-media.ts: Bodied Extension: Insert Media 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "bodied-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "default",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
      },
      "content": Array [
        Object {
          "attrs": Object {
            "layout": "center",
          },
          "content": Array [
            Object {
              "attrs": Object {
                "__fileName": "example.png",
                "collection": "MediaServicesSample",
                "height": 4096,
                "id": "1234-5678-abcd-efgh",
                "type": "file",
                "width": 4096,
              },
              "type": "media",
            },
          ],
          "type": "mediaSingle",
        },
        Object {
          "type": "paragraph",
        },
      ],
      "type": "bodiedExtension",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
